src/Framework/ModuleAssets.php

```diff
@@ -163,7 +163,10 @@ private function assetUrl(string $relativePath): string
             return $this->devBuildUrl . '/' . $assetPath->relativeTo($this->productionBuildPath);
         }
         $buildAssetPath = $this->buildAssetPath($assetPath);
-        return $buildAssetPath ? $buildAssetPath->url() : $relativePath;
+        if ($buildAssetPath) {
+            return $buildAssetPath->url();
+        }
+        return $assetPath->exists() ? $assetPath->url() : $relativePath;
```

Comment:
I don't know the details of the issue you were running into, but in general all asset urls should be built through the manifest while in a production build setting, and if not, they should fail loudly. Maybe returning an empty url string or something like /404 would be better? I don't think there's a scenario where loading the URL of the source asset is desirable in a production build setting.